api/src/pcapi/alembic/versions/20230810T101608_72fa9c444b5a_add_finance_incident_to_action_history.py

```diff
@@ -0,0 +1,41 @@
+"""Add finance incident to ActionHistory table
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# pre/post deployment: pre
+# revision identifiers, used by Alembic.
+revision = "72fa9c444b5a"
+down_revision = "117312f339eb"
+branch_labels = None
+depends_on = None
+
+fk_constaint_name = "action_history_incident_fkey"
+
+
+def upgrade() -> None:
+    op.add_column("action_history", sa.Column("financeIncidentId", sa.BigInteger(), nullable=True))
+    op.create_index(op.f("ix_action_history_financeIncidentId"), "action_history", ["financeIncidentId"], unique=False)
+    op.create_foreign_key(
+        fk_constaint_name, "action_history", "finance_incident", ["financeIncidentId"], ["id"], ondelete="CASCADE"
+    )
+
+    op.drop_constraint("check_at_least_one_resource_or_is_fraud_action", "action_history")
+    op.create_check_constraint(
+        "check_at_least_one_resource_or_is_fraud_action",
+        "action_history",
+        'num_nonnulls("userId", "offererId", "venueId", "financeIncidentId") >= 1 OR "actionType" = \'BLACKLIST_DOMAIN_NAME\' OR "actionType" = \'REMOVE_BLACKLISTED_DOMAIN_NAME\'',
+    )
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Je ne sais pas à quoi ça sert, je suis curieux.

Reply:
C'est la contrainte qui existait déjà avant sur le modèle ActionHistory, je l'ai adapté (drop and create) en y ajoutant `financeIncidentId`. Si aucun des quatres champs cités alors -> erreur

Reply:
Pour compléter la réponse de Charles, le but est de ne pas permettre d'enregistrer dans l'historique une action qui n'est rattachée à aucun objet.